### PR TITLE
Added trace_transaction

### PR DESCRIPTION
**Pull request by MadelineMurray**
Signed-off-by: Madeline Murray <madeline.taylor@gmail.com>

Added trace_transaction

### CODE REVIEW

File: docs/HowTo/Troubleshoot/Trace-Transactions.md [hunk 1]
```diff
@@ -10,13 +10,15 @@ To get detailed information about transaction processing, use the
...
+To specify the trace types required and retrieve the transaction traces for a block, use
+[`trace_replayBlockTransactions`](../../Reference/API-Methods.md#trace_replayblocktransactions). Options are
+[`trace`, `vmTrace`, or `stateDiff`](../../Reference/Trace-Types.md).
+
+To retrieve the [`trace` type](../../Reference/Trace-Types.md#trace) for a specific 
+transaction, use [`trace_transaction`](../../Reference/API-Methods.md#trace_transaction). 
 
 Your node must be an archive node (that is, synchronised without pruning or fast sync) or the
-requested block must be within the last 1024 blocks.
+requested block must be within the [the number of pruning blocks retained](../../Reference/CLI/CLI-Syntax.md#pruning-blocks-retained).
```
Comment (grantnoble) [inline docs/HowTo/Troubleshoot/Trace-Transactions.md]: Line length?

File: docs/Reference/API-Methods.md [hunk 2]
```diff
@@ -4386,12 +4386,12 @@ None
...
 ### trace_replayBlockTransactions
 
-Provides transaction processing tracing.
+Provides transaction processing tracing per block.
 
 !!! important
 
     Your node must be an archive node (that is, synchronised without pruning or fast sync) or the
-    requested block must be within the last 1024 blocks.
+    requested block must be within [the number of pruning blocks retained](../CLI/CLI-Syntax#pruning-blocks-retained).
```
Comment (grantnoble) [inline docs/Reference/API-Methods.md]: Line length?

File: docs/Reference/API-Methods.md [hunk 3]
```diff
@@ -4489,6 +4489,127 @@ one object per transaction, in transaction execution order.
...
     }
     ```
+### trace_transaction
+
+Provides transaction processing of [type `trace`](Trace-Types.md#trace) for the specified transction.
+
+!!! important
+
+    Your node must be an archive node (that is, synchronised without pruning or fast sync) or the
+    requested block must be within [the number of pruning blocks retained](../CLI/CLI-Syntax#pruning-blocks-retained).
```
Comment (grantnoble) [inline docs/Reference/API-Methods.md]: Line length?

**Comment by matkt** [comment]
LGTM

File: docs/HowTo/Troubleshoot/Trace-Transactions.md [hunk 4]
```diff
@@ -10,13 +10,18 @@ To get detailed information about transaction processing, use the
...
+
+To retrieve the [`trace` type](../../Reference/Trace-Types.md#trace) for a specific:  
+
+* Block, use [`trace_block`](../../Reference/API-Methods.md#trace_block). 
+* Transaction, use [`trace_transaction`](../../Reference/API-Methods.md#trace_transaction). 
+
 
 Your node must be an archive node (that is, synchronised without pruning or fast sync) or the
-requested block must be within the last 1024 blocks.
+requested block or transaction must be within the [the number of pruning blocks retained](../../Reference/CLI/CLI-Syntax.md#pruning-blocks-retained).
```
Comment (NicolasMassart) [inline docs/HowTo/Troubleshoot/Trace-Transactions.md]: Should we indicate the default pruning value here to help not navigating if user did not customise this?

File: docs/Reference/API-Methods.md (hunk 2 above)
Comment (NicolasMassart) [inline docs/Reference/API-Methods.md]: same about default value

File: docs/Reference/API-Methods.md [hunk 5]
```diff
@@ -4490,6 +4490,222 @@ one object per transaction, in transaction execution order.
...
     ```
 
+### trace_block
+
+Provides transaction processing of [type `trace`](Trace-Types.md#trace) for the specified block.
+
+!!! important
+
+    Your node must be an archive node (that is, synchronised without pruning or fast sync) or the
+    requested block must be within [the number of pruning blocks retained](../CLI/CLI-Syntax#pruning-blocks-retained).
```
Comment (NicolasMassart) [inline docs/Reference/API-Methods.md]: default value?

File: docs/Reference/API-Methods.md [hunk 6]
```diff
@@ -4490,6 +4490,222 @@ one object per transaction, in transaction execution order.
...
+
+### trace_transaction
+
+Provides transaction processing of [type `trace`](Trace-Types.md#trace) for the specified transction.
+
+!!! important
+
+    Your node must be an archive node (that is, synchronised without pruning or fast sync) or the
+    requested transaction must be contained in a blocked within 
+    [the number of pruning blocks retained](../CLI/CLI-Syntax#pruning-blocks-retained).
```
Comment (NicolasMassart) [inline docs/Reference/API-Methods.md]: default value?

File: mkdocs.yml [hunk 7]
```diff
@@ -233,3 +233,4 @@ plugins:
           HowTo/Deploy/Monitoring-Performance.md: HowTo/Monitor/Metrics.md
           HowTo/Upgrade/Upgrade-Network.md: HowTo/Upgrade/Upgrade-Node.md
           HowTo/Find-and-Connect/Using-UPnP.md: HowTo/Find-and-Connect/Specifying-NAT.md
+          Concepts/Transactions/Trace-Types.md: Reference/Trace-Types.md
```
Comment (NicolasMassart) [inline mkdocs.yml]: 👍

File: docs/HowTo/Troubleshoot/Trace-Transactions.md (hunk 4 above)
Comment (MadelineMurray) [reply docs/HowTo/Troubleshoot/Trace-Transactions.md]: I was in two minds about that.  Have added now.

File: docs/Reference/API-Methods.md (hunk 2 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: It's a link putting it over so I'm going to leave as is

Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: The block parameter isn't optional

File: docs/Reference/API-Methods.md (hunk 5 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Added

File: docs/Reference/API-Methods.md (hunk 6 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: Added

File: docs/Reference/API-Methods.md (hunk 3 above)
Comment (MadelineMurray) [reply docs/Reference/API-Methods.md]: It's a link putting it over so I'm going to leave it in this case

File: docs/HowTo/Troubleshoot/Trace-Transactions.md (hunk 1 above)
Comment (MadelineMurray) [reply docs/HowTo/Troubleshoot/Trace-Transactions.md]: It's a link putting it over so I'm going to leave it in this case

**Pull request merged**